app/assets/javascripts/discourse/adapters/topic.js.es6

```diff
@@ -1,7 +1,35 @@
 import { ajax } from 'discourse/lib/ajax';
 import RestAdapter from 'discourse/adapters/rest';
+import Topic from 'discourse/models/topic';
+
+export function Result(payload, responseJson) {
+  this.payload = payload;
+  this.responseJson = responseJson;
+  this.target = null;
+}
 
 export default RestAdapter.extend({
+  update(store, type, id, attrs) {
+    const data = {};
+    const typeField = Ember.String.underscore(type);
+    Object.assign(data, attrs);
+    return this.pathFor(store, type, id).then(function(path) {
+      return ajax(path, { method: 'PUT', data }).then(function(json) {
+        const result = new Result(json[typeField], json);
+        console.log(result);
+        return result;
+      });
+    });
+  },
+  pathFor(store, type, findArgs) {
+    var self = this;
+    return Topic.find(findArgs, {}).then(function(result) {
+      if (result) {
+        self.set('path', self.basePath(store, type, findArgs) + 't/' + result.slug);
```

Comment:
Can be written as `${self.basePath(store, type, findArgs)}/t/${result.slug}` with ES6's [template literals](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals).